packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -6,108 +6,86 @@
-    error L1_PROOF_LENGTH();
-    error L1_PROVER();
-    error L1_ZKP();
 
     function proveBlock(
         TaikoData.State storage state,
         TaikoData.Config memory config,
         AddressResolver resolver,
         uint256 blockId,
-        bytes[] calldata inputs
-    ) public {
-        // Check and decode inputs
-        if (inputs.length != 3) revert L1_INPUT_SIZE();
-        TaikoData.Evidence memory evidence = abi.decode(
-            inputs[0],
-            (TaikoData.Evidence)
+        bytes calldata evidenceBytes
+    ) internal {
+        TaikoData.ValidBlockEvidence memory evidence = abi.decode(
+            evidenceBytes,
+            (TaikoData.ValidBlockEvidence)
         );
 
-        // Check evidence
-        if (evidence.meta.id != blockId) revert L1_ID();
+        TaikoData.BlockMetadata memory meta = evidence.meta;
+        _checkMetadata(state, config, meta, blockId);
 
-        if (evidence.proofs.length != 3) revert L1_PROOF_LENGTH();
+        BlockHeader memory header = evidence.header;
+        if (
+            evidence.signalRoot == 0 ||
+            evidence.prover == address(0) ||
+            header.parentHash == 0 ||
+            header.beneficiary != meta.beneficiary ||
+            header.difficulty != 0 ||
+            header.gasLimit != meta.gasLimit + config.anchorTxGasLimit ||
+            header.gasUsed == 0 ||
+            header.timestamp != meta.timestamp ||
+            header.extraData.length != meta.extraData.length ||
+            header.mixHash != meta.mixHash
+        ) revert L1_INVALID_EVIDENCE();
+
+        for (uint i = 0; i < header.extraData.length; ) {
+            if (header.extraData[i] != meta.extraData[i])
+                revert L1_INVALID_EVIDENCE();
+            unchecked {
+                ++i;
+            }
+        }
 
-        IProofVerifier proofVerifier = IProofVerifier(
-            resolver.resolve("proof_verifier", false)
+        bytes32 instance = _getInstance(
+            evidence,
+            resolver.resolve("signal_service", false),
+            resolver.resolve(config.chainId, "signal_service", false)
         );
 
-        if (config.enableAnchorValidation) {
-            _proveAnchorForValidBlock({
-                config: config,
-                resolver: resolver,
-                proofVerifier: proofVerifier,
-                evidence: evidence,
-                anchorTx: inputs[1],
-                anchorReceipt: inputs[2]
-            });
-        }
-
-        // ZK-prove block and mark block proven to be valid.
         _proveBlock({
             state: state,
             config: config,
             resolver: resolver,
-            proofVerifier: proofVerifier,
-            evidence: evidence,
-            target: evidence.meta,
-            blockHashOverride: 0
+            blockId: blockId,
+            parentHash: header.parentHash,
+            snippet: Snippet(header.hashBlockHeader(), evidence.signalRoot),
```

Comment:
Snippet seems a bit of strange name, also a bit confusing to me why these specific values are combined together here in that struct.

Reply:
Please suggest a better name

Reply:
Perhaps something like `L2State` or `L2StateData`?

Reply:
These two pieces of data (signalRoot and blockHeader) are also synced from L1 to L2. So naming it as L2*** is not appreciate.

Reply:
`ChainState` or `ChainStateData`?

Reply:
https://github.com/taikoxyz/taiko-mono/pull/13265

Renamed in the above PR as `chainData`